lilypad/developer-resources/create-lilypad-module/building-your-module.md

```diff
@@ -0,0 +1,349 @@
+
+```python
+inputs = tokenizer(
+    input,
+    return_tensors="pt",
+    truncation=True,
+    padding=True,
+)
+
+with torch.no_grad():
+    logits = model(**inputs).logits
+
+predicted_class_id = logits.argmax().item()
+model.config.id2label[predicted_class_id]
+
+return output
+```
+
+All we need to do from here is set the `output` to the last line we pasted.
+
+```python
+output = model.config.id2label[predicted_class_id]
+
+return output
+```
+
+That's everything we'll need for the modules source code! You can build the Docker image with the new code.
+
+```sh
+python -m scripts.docker_build
+```
+
+### 2. Delete Code Block
+
+While the Docker image is being built (it can take a while), we still need to finish step 2 that the error in the console gave us earlier. Open the `run_module.py` script.
+
+Find the `TODO` comment and delete the code block underneath.
+
+```python
+# TODO: Remove the following print and sys.exit statements and create the module job.
+print(
+    "❌ Error: No job configured. Implement the module's job before running the module.",
+    file=sys.stderr,
+    flush=True,
+)
+print("\t1. Implement job module")
+print("\t\t👉 /src/run_inference.py")
+print("\t2. Delete this code block")
+print("\t\t👉 /scripts/run_module.py")
+sys.exit(1)
+```
+
+{% hint style="info" %}
+Since we are editing a script and not the `src` code that gets used in the Docker image we won't need to rebuild after making these changes.
+{% endhint %}
+
+Once the Docker image is finished building, we can run the module!
+
+## Running Your Module
+
+It's finally time to see your module in action.
+
+### Local
+
+Let's start by running it locally.
+
+```sh
+python -m scripts.run_module --local
+```
+
+The CLI should ask you for an input. Enter whatever you like and hit enter. The module will analyze the sentiment of your input and output the results at `outputs/result.json`.
+
+```json
+{
+  "input": "LILYPAD IS AWESOME",
+  "result": "POSITIVE",
+  "status": "success"
+}
+```
+
+You just used a local LLM! 🎉
+
+### Lilypad Network
+
+Before you can run the module on Lilypad Network, you'll need to push the Docker image to Docker Hub.
+
+```sh
+python -m scripts.docker_build --push
+```
+
+While the Docker image is being built and pushed, you should configure the rest of the variables in `constants.py`. Make sure that you push your code to a public GitHub repository.
+
+{% hint style="info" %}
+Since these variables are only used in scripts and not in any `src` code that gets used in the Docker image we won't need to rebuild after making these changes.
+{% endhint %}
+
+Once your Docker image is pushed to Docker Hub, your most recent code is pushed to a public GitHub repository, you can run your module on Lilypad Network by removing the `--local` flag.
+
+```sh
+python -m scripts.run_module
+```
+
+You just use an LLM on Lilypad's decentralized network! 🎉
```

Comment:
I love the simplicity of this conclusion but I would like to see instructions on how a user would run this also. Just a suggestion but maybe its worth showing the CLI instructions?